extensions/src/main/java/org/pentaho/platform/plugin/action/ActionInvokerSystemListener.java

```diff
@@ -145,9 +145,10 @@ public Payload( String json ) {
       buildPayload( json, LocaleHelper.UTF_8 );
     }
 
-    public Response issueRequest( ) {
+    public Response issueRequest( ) throws IOException {
       ActionResource actionResource = new ActionResource();
-      return actionResource.invokeAction( ActionUtil.INVOKER_SYNC_VALUE, actionId, actionClass, actionUser, actionParams );
+      return actionResource.invokeAction( ActionUtil.INVOKER_SYNC_VALUE, actionId, actionClass, actionUser,
+        ActionParams.fromJson( actionParams ) );
```

Comment:
Since now the REST resource signature now accepts ActionParams object (which is already de-serialized) we need to de-serialize here before calling the resource. Any problem with this? (or am I missing your point?)